File: discord_autodelete.js
const { Client, GatewayIntentBits } = require('discord.js');

const client = new Client({
  intents: [
    GatewayIntentBits.Guilds,
    GatewayIntentBits.GuildMessages,
    GatewayIntentBits.MessageContent,
  ],
});

// Replace with your channel ID (just the numbers)
const TARGET_CHANNEL_ID = '1425176266293645363';
const DELETE_DELAY = 15000; // 15 seconds in milliseconds

client.on('ready', () => {
  console.log(`✅ Bot logged in as ${client.user.tag}`);
  console.log(`🗑️  Auto-deleting messages in channel ${TARGET_CHANNEL_ID} after 15 seconds`);
});

client.on('messageCreate', async (message) => {
  // Only process messages in the target channel
  if (message.channel.id !== TARGET_CHANNEL_ID) return;
  
  // Don't delete bot's own messages or messages that are already being deleted
  if (message.author.bot) return;

  try {
    // Wait 15 seconds then delete the message
    setTimeout(async () => {
      try {
        await message.delete();
        console.log(`🗑️  Deleted message from ${message.author.tag}`);
      } catch (error) {
        // Message might already be deleted or bot lacks permissions
        console.error('Error deleting message:', error.message);
      }
    }, DELETE_DELAY);
  } catch (error) {
    console.error('Error setting up deletion:', error);
  }
});

// Login with your bot token
client.login(process.env.DISCORD_BOT_TOKEN);

File: discord_autodelete.py
import discord
import asyncio
import os

# Bot setup
intents = discord.Intents.default()
intents.message_content = True
client = discord.Client(intents=intents)

# Configuration
TARGET_CHANNEL_ID = 1425176266293645363  # Your channel ID
DELETE_DELAY = 15  # seconds

@client.event
async def on_ready():
    print(f'✅ Bot logged in as {client.user}')
    print(f'🗑️  Auto-deleting messages in channel {TARGET_CHANNEL_ID} after {DELETE_DELAY} seconds')

@client.event
async def on_message(message):
    # Only process messages in the target channel
    if message.channel.id != TARGET_CHANNEL_ID:
        return
    
    # Don't delete bot's own messages
    if message.author.bot:
        return
    
    # Wait for the specified delay, then delete
    await asyncio.sleep(DELETE_DELAY)
    
    try:
        await message.delete()
        print(f'🗑️  Deleted message from {message.author.name}')
    except discord.errors.NotFound:
        print(f'⚠️  Message already deleted')
    except discord.errors.Forbidden:
        print(f'❌ Missing permissions to delete message')
    except Exception as e:
        print(f'❌ Error: {e}')

# Run the bot
client.run(os.environ.get('DISCORD_BOT_TOKEN'))
